chef/cookbooks/ceph/recipes/calamari.rb

```diff
@@ -0,0 +1,55 @@
+node['ceph']['calamari']['packages'].each do |pkg|
+  package pkg
+end
+
+# TODO: do we need pacemaker sync marks? (nothing here uses pacemaker;
+# something for further consideration is making the calamari server HA)
+
+ruby_block "initialize calamari server" do
+  block do
+    check_cmd = Mixlib::ShellOut.new("/srv/www/calamari/manage.py check >/dev/null 2>&1")
+    check_cmd.run_command
+    if check_cmd.exitstatus != 0
+      Chef::Log.info("Ceph: Initializing Calamari")
+      # If check failed, it'll be because calamari isn't initialized yet, so
+      # set it up (newer versions of Mixlib::ShellOut have an error? method
+      # we could call instead of checking exitstatus, but this seems to be
+      # sadly missing on SLE at the moment)
+      init_cmd = Mixlib::ShellOut.new(
+        "calamari-ctl", "initialize",
+          "--admin-username", node[:ceph][:calamari][:username],
+          "--admin-password", node[:ceph][:calamari][:password],
+          "--admin-email", node[:ceph][:calamari][:email])
+      init_cmd.run_command
+      init_cmd.error!
+    else
+      # Check passed, so it's already configured.  Need to update admin user.
+      # Is it unholy to generate python from ruby?
+      Chef::Log.info("Ceph: Updating Calamari admin user")
+      update_cmd = Mixlib::ShellOut.new('/srv/www/calamari/manage.py shell', :input => <<eos)
```

Comment:
I'm happy to merge in the current state and address any remaining issues in a followup PR.

Improvements for deployment in this context probably belong in https://bugzilla.suse.com/ against SUSE Storage, either the Barclamp or Calamari component. From there I can triage and either fix if it's SUSE specific, or raise upstream if not.